src/main/java/com/github/wovnio/wovnjava/CustomDomainUrlLanguagePatternHandler.java

```diff
@@ -0,0 +1,66 @@
+package com.github.wovnio.wovnjava;
+
+import java.util.regex.Pattern;
+import java.net.URL;
+import java.net.MalformedURLException;
+
+class CustomDomainUrlLanguagePatternHandler extends UrlLanguagePatternHandler {
+    private Lang defaultLang;
+    private CustomDomainLanguages customDomainLanguages;
+
+    CustomDomainUrlLanguagePatternHandler(Lang defaultLang, CustomDomainLanguages customDomainLanguages) {
+        this.defaultLang = defaultLang;
+        this.customDomainLanguages = customDomainLanguages;
+    }
+
+    Lang getLang(String urlString) {
+        URL url = getUrlObject(urlString);
+        if (url == null) return null;
+
+        CustomDomainLanguage customDomainLanguage = this.customDomainLanguages.getCustomDomainLanguageByUrl(url);
+        if (customDomainLanguage == null) return null;
+
+        return customDomainLanguage.lang;
+    }
+
+    String convertToDefaultLanguage(String urlString) {
+        return convertToTargetLanguage(urlString, this.defaultLang);
+    }
+
+    String convertToTargetLanguage(String urlString, Lang lang) {
+        URL url = getUrlObject(urlString);
+        if (url == null) return urlString;
+
+        CustomDomainLanguage currentCDL = this.customDomainLanguages.getCustomDomainLanguageByUrl(url);
+        if (currentCDL == null) return urlString;
+
+        CustomDomainLanguage targetCDL = this.customDomainLanguages.getCustomDomainLanguageByLang(lang);
+        if (targetCDL == null || currentCDL == targetCDL) return urlString;
+
+        Pattern p = Pattern.compile("^(" + currentCDL.path + ")");
```

Comment:
No problems with trailing slashes etc with this regex? Haven't looked at the tests yet

Reply:
`CustomDomainLanguage.path` is always stored _without_ trailing slash. This is enforced by the Serializer when reading the configuration string. Maybe it should be enforced when initializing CustomDomainLanguage as well?

When we execute this regex, `CustomDomainLanguages.getCustomDomainLanguageByUrl` has already established that the paths match (followed by `/\/|$/`). So the regex just does a simple replace.

I believe the tests cover all the cases, but the various responsibilities are somewhat distributed among the `CustomDomainLanguage*` files.

Reply:
I think I did the opposite in wovn.cs, where CustomDomainLanguage.Path was guaranteed to end with a trailing slash.

Reply:
I ignored trailing slash to swap between `site.com` -> `site.com/ko` and back without additional logic for it, just as I would for `site.com/` -> `site.com/ko/`.